hpx/runtime/components/component_factory.hpp

```diff
@@ -166,9 +168,25 @@ namespace hpx { namespace components
         ///         sequential in a row.
         naming::gid_type create(std::size_t count = 1)
         {
+            return create<Component>(count,
+                std::is_default_constructible<
+                    typename Component::type_holder>());
+        }
+
+        template <typename Component_>
+        naming::gid_type create(std::size_t count, std::false_type)
+        {
+            // shouldn't ever be called for non-default-constructible types
+            HPX_ASSERT(false);
```

Comment:
No, always - this function is called by `create(size_t)` which is a virtual function.